src/Controller/WebPageArchiveController.php

```diff
@@ -53,4 +57,48 @@ public function title($web_page_archive) {
     return $web_page_archive->label();
   }
 
+  /**
+   * Common batch processing callback for all operations.
+   */
+  public static function batchProcess(WebPageArchive $web_page_archive, &$context) {
+    $queue = \Drupal::service('queue')->get("web_page_archive_capture.{$web_page_archive->uuid()}");
+    $queue_worker = \Drupal::service('plugin.manager.queue_worker')->createInstance('web_page_archive_capture');
+
+    // TODO: Move threshold into admin panel.
+    // Per-entity or global for all settings?
+    $number_of_queue = ($queue->numberOfItems() < WEB_PAGE_ARCHIVE_BATCH_SIZE) ? $queue->numberOfItems() : WEB_PAGE_ARCHIVE_BATCH_SIZE;
+
+    for ($i = 0; $i < $number_of_queue; $i++) {
+      // Claim item and attempt to process it.
+      if ($item = $queue->claimItem()) {
+        try {
+          $queue_worker->processItem($item->data);
+          $queue->deleteItem($item);
+        }
+        catch (SuspendQueueException $e) {
+          $queue->releaseItem($item);
+          watchdog_exception($e);
+          break;
```

Comment:
Good call on the `RequeueException`. Cron also has this as well:
```
catch (\Exception $e) {
  // In case of any other kind of exception, log it and leave the item
  // in the queue to be processed again later.
  watchdog_exception('cron', $e);
}
```

I have no idea what the reasons were for adding that, but it may not be a bad idea to add that  just in case.